tests/unit/Codeception/Step/RetryTest.php

```diff
@@ -0,0 +1,31 @@
+<?php
+namespace Codeception\Step;
+
+use Codeception\Lib\ModuleContainer;
+use Codeception\Util\Stub;
+
+class RetryTest extends \PHPUnit\Framework\TestCase
+{
+    protected $shouldFail = true;
+
+    public function testRetryStepShouldNotFailStep()
+    {
+        $moduleContainer = Stub::make(ModuleContainer::class, [
+            'moduleForAction' => $this
+        ]);
+        $retry = new \Codeception\Step\Retry('_executeFailedCode', [], 2, 0);
+        $retry->run($moduleContainer);
+
```

Comment:
I must admit that I can't understand retry tests, including older timing based tests, at all.
It would be good to have a detailed explanation in comments what actually happens here.

Reply:
Ok